datahub-classify/tests/test_similarity_predictor.py

```diff
@@ -0,0 +1,487 @@
+import glob
+import itertools
+import json
+import logging
+import os
+import pickle
+import time
+from typing import Dict, Tuple
+
+import numpy as np
+import pandas as pd
+import pytest
+from sklearn.metrics import classification_report, confusion_matrix
+
+from datahub_classify.helper_classes import (
+    ColumnInfo,
+    ColumnMetadata,
+    TableInfo,
+    TableMetadata,
+)
+from datahub_classify.similarity_predictor import check_similarity
+
+logger = logging.getLogger(__name__)
+logging.basicConfig(level=logging.INFO)
+logger.info("libraries Imported..................")
+
+SEED = 100
+np.random.seed(SEED)
+PRUNING_THRESHOLD = 0.8
+FINAL_THRESHOLD = 0.6
+COLUMN_SIMILARITY_THRESHOLD = 0.8
+PLATFORMS = ["A", "B", "C", "D", "E"]
+
+
+def load_df(dataset_name):
+    path = all_datasets_paths[dataset_name]
+    if path.endswith("csv"):
+        df = pd.read_csv(path, nrows=2)
+    elif path.endswith("xlsx"):
+        df = pd.read_excel(path, nrows=2)
+    else:
+        df = None
+    return df
+
+
+def load_jsons(input_jsons_dir):
+    with open(
+        os.path.join(input_jsons_dir, "table_similarity_labels_IDEAL.json")
+    ) as filename:
+        table_similarity_labels_ideal_ = json.load(filename)
+    with open(
+        os.path.join(input_jsons_dir, "pruning_table_similarity_labels_EXPECTED.json")
+    ) as filename:
+        pruning_table_similarity_labels_expected_ = json.load(filename)
+    with open(
+        os.path.join(
+            input_jsons_dir, "post_pruning_table_similarity_labels_EXPECTED.json"
+        )
+    ) as filename:
+        post_pruning_table_similarity_labels_expected_ = json.load(filename)
+
+    with open(
+        os.path.join(input_jsons_path, "column_similarity_scores_EXPECTED.json")
+    ) as filename_:
+        column_similarity_scores_expected_ = json.load(filename_)
+
+    with open(
+        os.path.join(input_jsons_path, "column_similarity_labels_IDEAL.json")
+    ) as filename_:
+        column_similarity_labels_ideal_ = json.load(filename_)
+
+    return (
+        table_similarity_labels_ideal_,
```

Comment:
why do we do this instead of reading each file in the test itself

Reply:
Restructured the unit testing code.